tools/hive-bigquery/bigquery_component.py

```diff
@@ -0,0 +1,725 @@
+        """Writes comparison metrics to BigQuery
+
+        Flattens the schema of both the Hive table and BigQuery table,
+        reads the data types validation rules list, does health checks of the
+        migration and loads the metrics data into a BigQuery comparison table
+
+        Args:
+            hive_component (:class:`HiveComponent`): Instance of
+                HiveComponent to connect to Hive
+            gcs_component (:class:`GCSStorageComponent`): Instance of
+            GCSStorageComponent to do GCS operations
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+        """
+
+        metrics_table_name = PropertiesReader.get('hive_bq_comparison_table')
+        metrics_csv_filename = PropertiesReader.get('hive_bq_comparison_csv')
+
+        print_and_log("Analyzing the Hive and BigQuery tables...")
+
+        # Flattens the Hive table schema and writes row to CSV file
+        hive_flat_schema, flat_list_columns = hive_component.flatten_schema(
+            hive_table_model)
+        hive_table_analysis = self.analyze_hive_table(hive_table_model,
+                                                      hive_flat_schema)
+        self.append_row_to_metrics_file(metrics_csv_filename,
+                                        hive_table_analysis, flat_list_columns)
+        logger.debug("Analyzed Hive table metrics")
+
+        # Flattens the BigQuery table schema and writes row to CSV file
+        bq_flat_schema = self.flatten_schema(bq_table_model)
+        bq_table_analysis = self.analyze_bq_table(bq_table_model,
+                                                  bq_flat_schema)
+        self.append_row_to_metrics_file(metrics_csv_filename, bq_table_analysis,
+                                        flat_list_columns)
+        logger.debug("Analyzed BigQuery table metrics")
+
+        logger.debug("Reading the validations CSV file")
+        validations_list = self.read_validations()
+        # Does Health checks by comparing Hive and BigQuery metrics
+        healths = self.do_health_checks(validations_list, hive_table_analysis,
+                                        bq_table_analysis, flat_list_columns)
+        self.append_row_to_metrics_file(metrics_csv_filename, healths,
+                                        flat_list_columns)
+        logger.debug("Health checks are done")
+
+        logger.debug("Getting metrics table schema")
+        metrics_table_schema = self.generate_metrics_table_schema(
+            flat_list_columns)
+
+        logger.debug("Creating BigQuery metrics table")
+        self.create_table(bq_table_model.dataset_id, metrics_table_name,
+                          metrics_table_schema)
+        # Uploads metrics CSV file to GCS bucket
+        blob_name = "BQ_staging/" + metrics_csv_filename
```

Comment:
This can be modified to:
blob_name = "BQ_staging/{}".format(metrics_csv_filename)